spec/lib/miq_automation_engine/models/miq_ae_git_import_spec.rb

```diff
@@ -0,0 +1,113 @@
+describe MiqAeGitImport do
+  before do
+    EvmSpecHelper.local_guid_miq_server_zone
+    @user = FactoryGirl.create(:user_with_group)
+  end
+
+  context "git import" do
+    let(:branch_name) { "b1" }
+    let(:tag_name) { "t1" }
+    let(:domain_name) { "BB8" }
+    let(:url) { "http://www.example.com/x/y" }
+    let(:domain) do
+      FactoryGirl.create(:miq_ae_git_domain,
+                         :tenant => @user.current_tenant,
+                         :name   => domain_name)
+    end
+    let(:repo) { FactoryGirl.create(:git_repository, :url => url) }
+    let(:branch_hash) do
+      {'ref' => branch_name, 'ref_type' => MiqAeDomain::BRANCH}
+    end
+
+    let(:tag_hash) do
+      {'ref' => tag_name, 'ref_type' => MiqAeDomain::TAG}
+    end
+
+    let(:branch) { FactoryGirl.create(:git_branch, :name => branch_name) }
+    let(:tag) { FactoryGirl.create(:git_tag, :name => tag_name) }
+
+    it "import a domain given a git repository and a branch" do
```

Comment:
@eclarizio I like the changes you suggested I would implement those shortly. Regarding too much going on, we have other specs that cover the import aspects from a Git filesystem, also there are separate specs for Repository . This is just a wrapper class to move functions out of MiqAeDomain class, eventually we would be removing the import logic in MiqAeDomain and use the new MiqAeImport class with the correct parameters so that if someone passes in git_url or git_repository_id we can call MiqAeGitImport.